File: app/src/main/java/eu/alexanderfischer/dvbverspaetungsinfo/helper/TextHelper.java
package eu.alexanderfischer.dvbverspaetungsinfo.helper;

import android.content.Context;
import android.content.SharedPreferences;
import android.preference.PreferenceManager;

import eu.alexanderfischer.dvbverspaetungsinfo.models.Delay;
import io.realm.RealmList;

/**
 * Created by Alexander Fischer.
 *
 * Helper class for doing text transformation stuff and adding information to DelayInformation object.
 */
public class TextHelper {

    public static String makeInfoText(Context context, Delay delay, boolean isFilterActivated) {
        String infoText = "";

        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
        boolean hasActivatedNotificationForLine = false;

        if (delay.getLinien().size() > 0) {
            switch (delay.getState()) {
                case "negativ":
                    infoText = "Störungsmeldung für die ";
                    break;
                case "positiv":
                    infoText = "Entwarnung für die ";
                    break;
                default:
                    infoText = "Hinweis für die ";
                    break;
            }

            RealmList<String> linien = delay.getLinien();
            RealmList<String> filteredLinien = new RealmList<>();

            if (isFilterActivated) {
                for (String linie : delay.getLinien()) {
                    hasActivatedNotificationForLine = sharedPref.getBoolean("linie" + linie, false);

                    if (hasActivatedNotificationForLine) {
                        filteredLinien.add(linie);
                    }
                }

                if (filteredLinien.size() > 0) {
                    linien = filteredLinien;
                }
            }

            if (linien.size() == 1) {
                String linie = linien.get(0);

                infoText = infoText + "Linie " + linie + ".";
            } else {
                StringBuilder linienText = new StringBuilder();

                infoText = infoText + "Linien ";
                for (int i = 0; i < linien.size(); i++) {

                    String linie = linien.get(i);

                    if (i == (linien.size() - 1)) {
                        linienText.append(linie);
                    } else {
                        linienText.append(linie).append(", ");
                    }

                }
                infoText = infoText + linienText.toString() + ".";
            }
        }

        return infoText;
    }

}

File: app/src/main/java/eu/alexanderfischer/dvbverspaetungsinfo/helper/TextHelper.kt
package eu.alexanderfischer.dvbverspaetungsinfo.helper

import eu.alexanderfischer.dvbverspaetungsinfo.models.Delay
import java.text.SimpleDateFormat
import java.util.*

/**
 * Created by Alexander Fischer.
 *
 * Helper class for doing text transformation stuff and adding information to DelayInformation object.
 */
object TextHelper {

    fun makeInfoText(delay: Delay): String {
        var infoText = ""

        if (delay.linien.size > 0) {
            infoText = when (delay.state) {
                "negativ" -> "Störungsmeldung für die "
                "positiv" -> "Entwarnung für die "
                else -> "Hinweis für die "
            }

            val linien = delay.linien

            if (linien.size == 1) {
                val linie = linien[0]

                infoText = infoText + "Linie " + linie + "."
            } else {
                val linienText = StringBuilder()

                infoText += "Linien "
                for (i in linien.indices) {

                    val linie = linien[i]

                    if (i == linien.size - 1) {
                        linienText.append(linie)
                    } else {
                        linienText.append(linie).append(", ")
                    }

                }
                infoText = infoText + linienText.toString() + "."
            }
        }

        return infoText
    }

    fun makeDayText() {
        val currentDate = DateHelper.currentTimeAndDate
        val currentDayOfWeek = SimpleDateFormat("EE", Locale.ENGLISH).format(currentDate.time)
    }

}
